services/frontend-service/src/ui/utils/store.tsx

```diff
@@ -128,12 +128,20 @@ export const refreshTags = (): void => {
 };
 export const [useTag, updateTag] = createStore<TagsResponse>({ response: tagsResponse, tagsReady: false });
 
-export const getCommitInfo = (commitHash: string, authHeader: AuthHeader): void => {
+export const getCommitInfo = (
+    commitHash: string,
+    pageNumber: number,
+    pageSize: number,
+    authHeader: AuthHeader
+): void => {
     useApi
         .gitService()
-        .GetCommitInfo({ commitHash: commitHash }, authHeader)
+        .GetCommitInfo({ commitHash: commitHash, pageNumber: pageNumber, pageSize: pageSize }, authHeader)
         .then((result: GetCommitInfoResponse) => {
-            updateCommitInfo.set({ response: result, commitInfoReady: CommitInfoState.READY });
+            var requestResult: GetCommitInfoResponse = structuredClone(result);
+            var oldEvents = updateCommitInfo.get().response?.events.slice() ?? [];
```

Comment:
```suggestion
            var requestResult: GetCommitInfoResponse = structuredClone(result);
            var oldEvents = updateCommitInfo.get().response?.events.slice() ?? [];
```
these should be const.